operation/operation.h

```diff
@@ -6,57 +6,70 @@
 #include "../Lora/Lora.h"
 #include "../eeprom/eeprom.h"
 
-#define TIME_S ((unsigned long) time_us_64() / 1000000)
+// Desi-Second, used in get_time_with_decimal
+#define TIME_DS (time_us_64() / 100000)
+#define DEC_OFFSET 10 // decimal offset
+#define SECS_IN_MIN 60 // seconds in a minute
+#define SECS_IN_HOUR 3600 // seconds in an hour
+#define TIMESTAMP_LEN 9
+
+#define TIME_S (time_us_64() / 1000000)
+
 #define PILL_INTERVAL_S 10
-#define MAX_COMP_COUNT 7 // comp = compartment
+#define PILLCOMP_COUNT 7 // pillcomp = compartments for pills
+#define OPTO_OFFSET 10
 #define BLINK_COUNT 5
 
+
+
 // LoRa doesn't like \n's ; considers them as operators
-static const char * log_msg[] =
+static const char * log_format[STRLEN_EEPROM - 1 - TIMESTAMP_LEN] =
         {" Boot",
          " LoRa connection established",
          " LoRa connection failed...",
-         " Switch #%u pressed",
+         " Waiting for switch press...",
+         " SW_%u pressed",
          " Calibrating...",
-         " Calibration finished! Full revolution: %d steps",
+         " Calibration finished! Full revolution: %hu steps",
          " Calibration finished! At compartment #%hhu",
          " Dispensing starting from compartment #%hhu",
          " Dispensing finished! Pills detected: %hhu",
          " Rotating to compartment %hhu...",
-         " Rotation finished",
-         " Pill found in compartment %d",
-         " No pill found in compartment %d, blink lights"};
+         " Rotation finished; now at compartment %hhu",
+         " Pill found in compartment %hhu",
+         " No pill found in compartment %hhu, blink lights"};
 
 // used most crucially for logf_msg switch case
 enum logs {
     BOOT,
-    LORA_1,
-    LORA_0,
+    LORA_CONNECTED,
+    LORA_NOT_CONNECTED,
+    WAITING_FOR_SW,
     SW_PRESSED,
-    CALIB_1,
-    CALIB_0_1,
-    CALIB_0_2,
-    DISP_1,
-    DISP_0,
-    ROT_1,
-    ROT_0,
-    PILL_1,
-    PILL_0,
+    CALIBRATING_STARTED,
+    CALIBRATED_FULL,
+    CALIBRATED_PARTIAL,
+    DISPENSING_STARTED,
+    DISPENSING_FINISHED,
+    COMPARTMENT_ROTATION_STARTED,
+    COMPARTMENT_ROTATION_FINISHED,
+    PILL_DETECTED,
+    PILL_NOT_DETECTED,
```

Comment:
I've uploaded another commit that clarify this part. We're now redo the work. plz check my version before make any further adjustment.